### ISSUE TITLE
Forge fails to find successfully installed dependency

### ISSUE THREAD

**Issue by jmcph4** [gakonst/foundry#814]
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.1.0 (9ef25c2 2022-02-26T00:26:30.551716007+00:00)

### What command(s) is the bug in?

forge build

### Operating System

Linux

### Describe the bug

# Actual Behaviour #
```
$ pwd
/home/foobar/path/to/my/project
$ forge build
compiling...
Unable to resolve imported file: "set-protocol/protocol/modules/BasicIssuanceModule.sol"
Compiling 5 files with 0.8.12
Compilation finished successfully
Error: 
   0: Compiler run failed
      ParserError: Source "set-protocol/protocol/modules/BasicIssuanceModule.sol" not found: File not found. Searched the following locations: "".
       --> /home/foobar/path/to/my/project/src/MyContract.sol:7:1:
        |
      7 | import "set-protocol/protocol/modules/BasicIssuanceModule.sol";
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

      
   0: 

Location:
   cli/src/cmd/mod.rs:87

Backtrace omitted.
Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
```

## Context ##
Installed dependencies via:

```
$ forge install SetProtocol/set-protocol-v2-contracts-legacy
```
Relevant state of filesystem after successful completion:

```
$ pwd
/home/foobar/path/to/my/project
$ ls lib
ds-test  openzeppelin-contracts  set-protocol-v2-contracts-legacy
$ ls lib/set-protocol-v2-contracts-legacy
contracts  hardhat.config.ts  package-lock.json		README.md  test		      tsconfig.json  utils
external   package.json       pull_request_template.md	scripts    tsconfig.dist.json  tslint.json    yarn.lock
$ stat lib/set-protocol-v2-contracts-legacy/contracts/protocol/modules/BasicIssuanceModule.sol 
  File: lib/set-protocol-v2-contracts-legacy/contracts/protocol/modules/BasicIssuanceModule.sol
  Size: 9038      	Blocks: 24         IO Block: 4096   regular file
Device: 24h/36d	Inode: 727178      Links: 1
Access: (0644/-rw-r--r--)  Uid: ( 1000/  foobar)   Gid: ( 1000/  foobar)
Access: 2022-02-26 13:26:33.435977916 +1000
Modify: 2022-02-26 13:26:33.435977916 +1000
Change: 2022-02-26 13:26:33.435977916 +1000
 Birth: 2022-02-26 13:26:33.435977916 +1000
```

I then added a remapping manually, like so:

```
$ cat foundry.toml
[default]
src = 'src'
out = 'out'
libs = ['lib']
remappings = [
    'ds-test/=lib/ds-test/src/',
    'openzeppelin-contracts/=lib/openzeppelin-contracts/contracts',
    'set-contracts/=lib/set-protocol-contracts-v2-legacy/contracts'
]

# See more config options https://github.com/gakonst/foundry/tree/master/config
```

Then I attempted a build and the error occurred.

# Expected Behaviour #

Specified import to be found.

**Comment by cleanunicorn** [comment]
Could it be the problem that you remap `set-contracts`

```
$ cat foundry.toml
[default]
src = 'src'
out = 'out'
libs = ['lib']
remappings = [
    'ds-test/=lib/ds-test/src/',
    'openzeppelin-contracts/=lib/openzeppelin-contracts/contracts',
    'set-contracts/=lib/set-protocol-contracts-v2-legacy/contracts'
]

# See more config options https://github.com/gakonst/foundry/tree/master/config
```

But try to import a different path `set-protocol`.

```
$ pwd
/home/foobar/path/to/my/project
$ forge build
compiling...
Unable to resolve imported file: "set-protocol/protocol/modules/BasicIssuanceModule.sol"
Compiling 5 files with 0.8.12
Compilation finished successfully
Error: 
   0: Compiler run failed
      ParserError: Source "set-protocol/protocol/modules/BasicIssuanceModule.sol" not found: File not found. Searched the following locations: "".
       --> /home/foobar/path/to/my/project/src/MyContract.sol:7:1:
        |
      7 | import "set-protocol/protocol/modules/BasicIssuanceModule.sol";
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

      
   0: 

Location:
   cli/src/cmd/mod.rs:87

Backtrace omitted.
Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
```

I am saying you remap `set-contracts` but try to import `set-protocol`.

**Comment by jmcph4** [comment]
> Could it be the problem that you remap `set-contracts`
> 
> ```
> $ cat foundry.toml
> [default]
> src = 'src'
> out = 'out'
> libs = ['lib']
> remappings = [
>     'ds-test/=lib/ds-test/src/',
>     'openzeppelin-contracts/=lib/openzeppelin-contracts/contracts',
>     'set-contracts/=lib/set-protocol-contracts-v2-legacy/contracts'
> ]
> 
> # See more config options https://github.com/gakonst/foundry/tree/master/config
> ```
> 
> But try to import a different path `set-protocol`.
> 
> ```
> $ pwd
> /home/foobar/path/to/my/project
> $ forge build
> compiling...
> Unable to resolve imported file: "set-protocol/protocol/modules/BasicIssuanceModule.sol"
> Compiling 5 files with 0.8.12
> Compilation finished successfully
> Error: 
>    0: Compiler run failed
>       ParserError: Source "set-protocol/protocol/modules/BasicIssuanceModule.sol" not found: File not found. Searched the following locations: "".
>        --> /home/foobar/path/to/my/project/src/MyContract.sol:7:1:
>         |
>       7 | import "set-protocol/protocol/modules/BasicIssuanceModule.sol";
>         | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 
>       
>    0: 
> 
> Location:
>    cli/src/cmd/mod.rs:87
> 
> Backtrace omitted.
> Run with RUST_BACKTRACE=1 environment variable to display it.
> Run with RUST_BACKTRACE=full to include source snippets.
> ```
> 
> I am saying you remap `set-contracts` but try to import `set-protocol`.

Definitely a huge facepalm moment for me!

**TL;DR**: You're right.

Two issues:

 - In the contract itself, I'm trying to import from `set-protocol` when my remappings need `set-contracts`
 - The remappings themselves have a typo in the path: they say `lib/set-protocol-contracts-v2-legacy/contracts` when they **should** say `lib/set-protocol-v2-contracts-legacy/contracts/protocol/modules/BasicIssuanceModule.sol `

---

I suppose the initial error message threw me due to the search path reported by `forge` being empty, but I'm very hesitant to blame the error output when it was my own typo.
